src/cms/utils/slug_utils.py

```diff
@@ -13,6 +13,34 @@
 
 
 def generate_unique_slug(form_object, foreign_model):
+    """
+    This function acts like an interface and extracts all parameters of the form_object before actually generating
+    the unique slug, so that unique slug generation can be performed without any cleaned form data after a form submission.
+
+    :param form_object: The form which contains the slug field
+    :type form_object: ~django.forms.Form
+
+    :param foreign_model: If the form instance has a foreign key to another model (e.g. because it is a translation of
+                          a content-object), this paramaeter contains the model of the foreign related object.
+    :type foreign_model: ~django.db.models.Model
+
+    :raises ~django.core.exceptions.ValidationError: When no slug is given and there is also no field which can be used
+                                                     as fallback (either ``title`` or ``name``).
+
+    :return: An unique slug identifier
+    :rtype: str
+    """
+    kwargs = {
+        "region": form_object.region,
+        "language": form_object.language,
+        "slug": form_object.cleaned_data["slug"],
+        "manager": form_object.Meta.model.objects,
+        "page_translation": form_object.instance,
+    }
+    return generate_unique_slug_helper(foreign_model, **kwargs)
+
+
+def generate_unique_slug_helper(foreign_model, **kwargs):
```

Comment:
No, there is no reason to exclude `foreign_model` from kwargs, treating the parameters equally sounds reasonable to me
I will also exchange the function names, my choice was just because of the compatibility for all "users" of that function